test/e2e/common/node/container_probe.go

```diff
@@ -246,6 +246,25 @@ var _ = SIGDescribe("Probing container", func() {
 		runReadinessFailTest(f, pod, time.Minute)
 	})
 
+	/*
+		Release: v1.21
+		Testname: Pod liveness probe, container exec timeout, restart
+		Description: A Pod is created with liveness probe with a Exec action on the Pod. If the liveness probe call does not return within the timeout specified, liveness probe MUST restart the Pod. When ExecProbeTimeout feature gate is disabled, the timeout is ignored BUT a failing liveness probe MUST restart the Pod.
```

Comment:
will this change how Containerd behaves? Before, in 1.19, the timeout wasn't respected and the probe failing after the timeout, will not restart the pod. Is it intentional to fix this behavior?

Reply:
I think the easiest way to think of this is that `ExecProbeTimeout=false` is a post-1.20 solution that is only appropriate for dockershim configuration.

This PR does not intend to fix the containerd scenario. For containerd scenarios (all non-dockershim scenarios, probably), we should not recommend fiddling with ExecProbeTimeout.

Should we make that more clear in the comments?

Reply:
> Should we make that more clear in the comments?

Yes, I think so

Reply:
yes, comment will suffice. So in 1.20 the behavior for containerd has changed - now it respects the failed probe after timeout. This is a breaking change, but I'd say it is not that critical.